### ISSUE TITLE
app-arch/zstd-1.4.5 fails test with ${GRAPHITE} flags

### ISSUE THREAD

**Issue by jiblime** [InBetweenNames/gentooLTO#586]
**allButGraphite-zstd-1.4.5.log -- Pass** [allButGraphite-zstd-1.4.5.log](https://github.com/InBetweenNames/gentooLTO/files/5131175/allButGraphite-zstd-1.4.5.log)
`*FLAGS='-O3 -march=native -mtune=native -fipa-pta -fdevirtualize-at-ltrans -fno-semantic-interposition -flto -flto-compression-level=9 -flimit-function-alignment -falign-functions=32 -malign-data=cacheline -pipe'`
`LDFLAGS='-Wl,-O1 -Wl,--as-needed'`

**justGraphite-zstd-1.4.5.log -- Fail** [justGraphite-zstd-1.4.5.log](https://github.com/InBetweenNames/gentooLTO/files/5131176/justGraphite-zstd-1.4.5.log)
`FLAGS='-O2 -fgraphite-identity -floop-nest-optimize -march=native -pipe'`
`LDFLAGS='-Wl,-O1 -Wl,--as-needed'`

**allPlusGraphite-zstd-1.4.5.log -- Fail** [allPlusGraphite-zstd-1.4.5.log](https://github.com/InBetweenNames/gentooLTO/files/5131177/allPlusGraphite-zstd-1.4.5.log)
`*FLAGS='-O2 -fgraphite-identity -floop-nest-optimize -march=native -pipe -flto -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans'`
`LDFLAGS='-Wl,-O1 -Wl,--as-needed'`

```
zstd: ../lib/compress/../common/bitstream.h:214: BIT_addBitsFast: Assertion `(value>>nbBits) == 0' failed.
```

**Comment by Althorion** [comment]
Can’t reproduce—works for me.

**Comment by jiblime** [comment]
That's not good for me. Just making sure, this was with USE=test, correct? If so...I'm due for a world rebuild anyway haha

**Comment by barolo** [comment]
Can't reproduce either

**Comment by jiblime** [comment]
Try `USE=test ebuild /var/db/repos/gentoo/app-arch/zstd/zstd-1.4.5.ebuild clean test`

**Comment by barolo** [comment]
doesn't build that way

**Comment by jiblime** [comment]
Thanks for checking it out. Same here, I found the culprit: 
```Warning : using stdout as default output. Do not rely on this behavior: use explicit `-c` instead !```

It'd take some time to figure out the proper test parameters to verify if graphite is the real cause. I'm not all that crazy about that at the moment since I think removing only graphite is not the worst thing.

**Comment by barolo** [comment]
zstd is becoming quite quintessential, it's worth checking. I've noticed some irregularities in its behavior recently, so I'm going to set it on the safer side.

**Comment by jiblime** [comment]
Yeah, you're right on that. I'll be dialing back my flags on it as well.

So because of `stdout as default output`, the logs are considered `data` instead of `UTF-8 Unicode text [...]` or `ASCII text [...]`

I'm unsure of the the severity of this, but since zstd doesn't officially support USE=test I think the first step would be to customize the test phase to exclude that option before asking
